src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/CompletionListOptimizer.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using AliasedVSCommitCharacters = Microsoft.VisualStudio.LanguageServer.Protocol.SumType<string[], Microsoft.VisualStudio.LanguageServer.Protocol.VSInternalCommitCharacter[]>;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Completion
+{
+    internal static class CompletionListOptimizer
```

Comment:
We've been building a ton of services recently; however, I felt like this didn't really warrant "servicability". Feel free to object though 😄

Reply:
I think this is fine. IMO services are only needed when there are dependencies.